spec/unit/installer_spec.rb

```diff
@@ -103,46 +103,71 @@ def @installer.clean_pod_sources
         @installer.install!
       end
 
-      it 'runs source provider hooks before analyzing' do
-        @installer.unstub(:resolve_dependencies)
-        @installer.stubs(:validate_build_configurations)
-        @installer.stubs(:clean_sandbox)
-        def @installer.run_source_provider_hooks
-          @hook_called = true
-        end
-
-        def @installer.analyze(*)
+      describe 'handling spec sources' do 
+        before do 
+          @hooks_manager = Pod::HooksManager
+          @hooks_manager.instance_variable_set(:@registrations, nil)
+        end
+
+        it 'runs source provider hooks before analyzing' do
+          @installer.unstub(:resolve_dependencies)
+          @installer.stubs(:validate_build_configurations)
+          @installer.stubs(:clean_sandbox)
+          @installer.stubs(:analyze)
+          @installer.stubs(:run_source_provider_hooks).with do
+            @hook_called = true
+          end
+          @installer.install!
           @hook_called.should.be.true
         end
-        @installer.install!
-      end
 
-      it 'includes sources from source provider plugins' do
-        plugin_name = 'test-plugin'
-        Pod::HooksManager.register(plugin_name, :source_provider) do |context, options|
```

Comment:
This hook was lingering beyond the scope of this test. Only myself to blame since I'm the one who originally wrote this code, hah